src/dodal/devices/eurotherm.py

```diff
@@ -0,0 +1,120 @@
+from bluesky.protocols import Locatable, Location
+from ophyd_async.core import (
+    AsyncStatus,
+    StandardReadable,
+    StandardReadableFormat,
+    StrictEnum,
+)
+from ophyd_async.epics.core import epics_signal_r, epics_signal_rw, epics_signal_rw_rbv
+
+"""
+Note: See i11 cyberstar blower for implementation of Eurotherm Controller
+"""
+
+
+class AutotuneControl(StrictEnum):
+    OFF = "Off"
+    ON = "On"
+
+
+class ManualMode(StrictEnum):
+    AUTO = "Automatic"
+    MANUAL = "Manual"
+
+
+class EurothermUpdate(StrictEnum):
+    PASSIVE = "Passive"
+    EVENT = "Event"
+    IO = "I/O Intr"
+    S10 = "10 second"
+    S5 = "5 second"
+    S2 = "2 second"
+    S1 = "1 second"
+    S0_5 = ".5 second"
+    S0_2 = ".2 second"
+    S0_1 = ".1 second"
+
+
+class EurothermGeneral(StandardReadable, Locatable):
+    """A base class for any eurotherm controller."""
+
+    def __init__(
+        self,
+        prefix: str,
+        name: str = "",
+        setpoint_suffix: str = "SP",
+        ramprate_suffix: str = "RR",
+        output_suffix: str = "O",
+        manual_suffix: str = "MAN",
+        temp_suffix: str = "PV:RBV",
+        rbv_suffix: str = ":RBV",
+        update=False,
+    ):
+        with self.add_children_as_readables(StandardReadableFormat.HINTED_SIGNAL):
+            self.temp = epics_signal_r(float, f"{prefix}{temp_suffix}")
+
+        with self.add_children_as_readables():
+            self.setpoint = epics_signal_rw_rbv(
+                float, f"{prefix}{setpoint_suffix}", rbv_suffix
+            )
+            self.ramprate = epics_signal_rw_rbv(
+                float, f"{prefix}{ramprate_suffix}", rbv_suffix
+            )
+            self.output = epics_signal_rw_rbv(
+                float, f"{prefix}{output_suffix}", rbv_suffix
+            )
+            self.mode = epics_signal_rw_rbv(
+                ManualMode, f"{prefix}{manual_suffix}", rbv_suffix
+            )
+
+            if update:
+                self.update = epics_signal_rw(EurothermUpdate, f"{prefix}UPDATE.SCAN")
```

Comment:
```suggestion
            if update:
                self.update = epics_signal_rw(EurothermUpdate, f"{prefix}UPDATE.SCAN")
            else:
                self.update = None
```

May be required for the type checking? May not anymore

Reply:
From a plan writing perspective, I think it's easier to reason about extending the `EurothermGeneral` to `UpdatingEurothermGeneral` - if you have a plan where you need to modify the `update` you pass in an `Updating...` else all plans can be written to take an instance of `EurothermGeneral`